internal/beater/otlp/stacktrace_processor.go

```diff
@@ -0,0 +1,64 @@
+package otlp
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/elastic/apm-data/model/modelpb"
+	"github.com/elastic/apm-server/internal/r8"
+)
+
+type MapFetcher interface {
+	Fetch(ctx context.Context, name, version string) ([]byte, error)
+}
+
+type CrashDeobfuscator interface {
+	Deobfuscate(stacktrace *[]*modelpb.StacktraceFrame, mapFile io.Reader) error
+}
+
+type StacktraceProcessor struct {
+	fetcher      MapFetcher
+	deobfuscator CrashDeobfuscator
+}
+
+// NewStacktraceProcessor returns a StacktraceProcessor.
+func NewStacktraceProcessor(fetcher MapFetcher) *StacktraceProcessor {
+	return &StacktraceProcessor{fetcher, crashDeobfuscatorFunc(r8.Deobfuscate)}
+}
+
+func (p StacktraceProcessor) ProcessBatch(ctx context.Context, batch *modelpb.Batch) error {
+	for _, event := range *batch {
+		if isMobileCrash(event) {
+			err := p.deobfuscate(ctx, event)
+			if err != nil {
+				return err
```

Comment:
I'm not sure if raising an error in case one event's deobfuscation fails is the recommended way to address this case, given that any unchecked events (if any) will not get deobfuscated because of it. I'm open to suggestions!